docs/csharp/language-reference/keywords/ref.md

```diff
@@ -115,13 +122,30 @@ When the caller stores the value returned by the `GetBookByTitle` method as a re
 
 [!code-csharp[csrefKeywordsMethodParams#6](~/samples/snippets/csharp/language-reference/keywords/in-ref-out-modifier/RefParameterModifier.cs#5)]
 
+## Ref struct types
+
+Adding the `ref` modifier to a `struct` declaration declares that instances of that type must be stack allocated. In other words, these types can never be created on the
+heap as a member of another class. The primary motivation for this feature was <xref:System.Span%601> and related structures. <xref:System.Span%601> may contain a managed pointer as one of its members, the other being the length of the span. It's implemented a bit differently because C# doesn't support pointers to managed memory outside of an unsafe context. Any write that changes the pointer and the length is not atomic. That means a <xref:System.Span%601> would be subject to out of range errors or other type safety violations were it not constrained to a single stack frame. In addition, putting a managed pointer on the GC heap typically crashes at JIT time. The language rules ensure the safe use of these types.
+
+The goal of keeping a `ref struct` type as a stack-allocated variable introduces several rules that the compiler enforces for all `ref struct` types.
```

Comment:
Difficult to comprehend. What about:
> Because instances of a `ref struct` type can be only stack-allocated, you cannot use them in the following scenarios: